cmd/ctl/pkg/util/util.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2020 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	apiutil "github.com/jetstack/cert-manager/pkg/api/util"
+	cmapiv1alpha2 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+)
+
+// FetchCertificateFromCR fetches the x509 certificate from a CR and stores the certificate in file specified by certFilename.
+// Assumes CR is ready, otherwise returns error.
+func FetchCertificateFromCR(req *cmapiv1alpha2.CertificateRequest, certFileName string,
+	ioStreams genericclioptions.IOStreams) error {
+	// If CR not ready yet, error
+	if !apiutil.CertificateRequestHasCondition(req, cmapiv1alpha2.CertificateRequestCondition{
+		Type:   cmapiv1alpha2.CertificateRequestConditionReady,
+		Status: cmmeta.ConditionTrue,
+	}) {
+		return errors.New("CertificateRequest is not ready yet, unable to fetch certificate")
+	}
```

Comment:
I wonder if we want this test, or just whether to check whether `Status.Certificate` is empty.

They should be equivalent, but the cmd in this case confirms that the certificate isn't empty. If that's the check we want
then maybe it's better to do that one here so that we don't write empty files if that ever happens and the caller doesn't check?